### PR TITLE
fix: dispatch underlying op for aliases

### PR DESCRIPTION
**Pull request by gforsyth**
`ops.Alias` was added in Ibis recently and the dispatcher was defaulting
to `ops.ValueOp` when it received one -- instead this added in a
specific Alias dispatch rule to pull out the underlying op.

xref ibis-project/ibis#3276

### CODE REVIEW

File: ibis_substrait/compiler/translate.py
Comment (cpcloud) [inline ibis_substrait/compiler/translate.py:444]: Should we do this in a backwards compatible way? E.g., importing `Alias` and only registering if the import succeeds?

Comment (gforsyth) [reply ibis_substrait/compiler/translate.py:444]: hmm, it's a good question.  I'm not sure what, if any, backwards compatibility guarantees we're prepared to make at this point.  We could make it backwards compatible (at least for now) just to demonstrate that we can, I suppose.  
I realize I'm not even certain if `ibis-substrait` is compatible with ibis 2.1.1

Comment (gforsyth) [reply ibis_substrait/compiler/translate.py:444]: Roger that.  I'll add in some version gating

**Comment by gforsyth** [comment]
@cpcloud -- are the "expected checks" here not getting picked up because I changed the build matrix?  Not sure what they are waiting on

**Review by cpcloud — APPROVED** [review]
@gforsyth Yep, that's right. This LGTM. I'll merge this after #185.

**Comment by cpcloud** [comment]
Will adjust the required checks after merging this.

File: pyproject.toml
Comment (cpcloud) [inline pyproject.toml:23]: Does it make sense to make this `>=2,<4`?

File: ibis_substrait/compiler/decompile.py
Comment (cpcloud) [inline ibis_substrait/compiler/decompile.py:733]: Can we remove the `inflection` dependency in this PR?

File: pyproject.toml
Comment (gforsyth) [reply pyproject.toml:23]: yes -- I had this hard-pinned because of the now-revoked versions that were misreporting `__version__`

File: ibis_substrait/compiler/decompile.py
Comment (gforsyth) [reply ibis_substrait/compiler/decompile.py:733]: done

**Pull request merged**
